### PR TITLE
Send user network finish event

### PR DESCRIPTION
**Pull request by MDrakos**
https://www.pivotaltracker.com/story/show/174606570

Addresses AC:
 - We send the finish event with the value: "user_network"

### CODE REVIEW

File: installers/msi-language/StateDeploy/CustomAction.cs [hunk 1]
```diff
@@ -567,6 +567,17 @@ public static ActionResult GAReportUserExit(Session session)
...
         }
 
+        /// <summary>
+        /// Reports a user network error event to google analytics
+        /// </summary>
+        [CustomAction]
+        public static ActionResult GAReportUserNetwork(Session session)
+        {
+            session.Log("sending user network error event");
+            TrackerSingleton.Instance.TrackEventSynchronously(session, session["SESSION_ID"], "stage", "finished", "network", session["ProductVersion"]);
```
Comment (Naatan) [inline installers/msi-language/StateDeploy/CustomAction.cs:577]: the AC says `user_network`.

**Review by Naatan — CHANGES_REQUESTED** [review]

**Review by Naatan — APPROVED** [review]
I was just reviewing the AC, would still want Martin's eyes on this before we merge.

File: installers/msi-language/StateDeploy/CustomAction.cs [hunk 2]
```diff
@@ -616,6 +627,7 @@ public static ActionResult SetNetworkErrorProperties(Session session)
                 RollbarReport.Error(msg, session);
             }
 
+            session.DoAction("GAReportUserNetwork");
```
Comment (mdrohmann) [inline installers/msi-language/StateDeploy/CustomAction.cs:630]: Interesting!  So, you can run "immediate" actions from a "deferred" action, and then use the `session` variables.  Maybe we can use to reduce the list of variables that we set in the `CustomActionData`?

My concern with this is, that with this set-up both the `stage/finished/network` and the `stage/finished/failure` messages will be sent. @Naatan Is that ok?

**Review by mdrohmann — CHANGES_REQUESTED** [review]
I believe that in case of a network error, we are still sending the `finished/failure` event.  That may be okay.  But the cleaner solution (although of course very difficult one, might be to change the `GAReportFailure` CustomAction and check if we have to send a failure or a network event...

File: installers/msi-language/StateDeploy/CustomAction.cs (hunk 2 above)
Comment (MDrakos) [reply installers/msi-language/StateDeploy/CustomAction.cs:630]: This method is actually run as immediate. The flow is a little different, it's declared [here](https://github.com/ActiveState/cli/blob/master/installers/msi-language/Product.p.wxs#L224) and eventually called via the Custom fatal error dialog [here](https://github.com/ActiveState/cli/blob/master/installers/msi-language/CustomFatalError.wxs#L37). I had to do it this way as you can only schedule one action to occur with `OnExit="error"`

Comment (mdrohmann) [reply installers/msi-language/StateDeploy/CustomAction.cs:630]: Ah! Maybe I get it know.  So, `OnExit=error` we are just invoking the `SetNetworkErrorProperties` custom action which then is calling the `CustomFatalError` dialog.

However, the `CustomFatalError` dialog is still calling `GAReportFailure` when the user clicks the `"Finish"` button https://github.com/ActiveState/cli/blob/302fe08257775318e0602bd5348bb1b8024df085/installers/msi-language/Product.p.wxs#L130.

But don't we then invoke `GAReportUserNetwork` every time, even if `NETWORK_ERROR` is empty?

Comment (MDrakos) [reply installers/msi-language/StateDeploy/CustomAction.cs]: Yes you are right about the `GaReportUserNetwork`. That should be now fixed.

I've decoupled the `GAReportFailure` from the Finish button on the exit dialog and instead it is now triggered from `SetNetworkProperties` if there is no network error.

**Review by mdrohmann — APPROVED** [review]

**Pull request merged**